addons/account/models/account_bank_statement.py

```diff
@@ -125,7 +125,11 @@ class AccountBankStatement(models.Model):
     # won't be called and therefore the other field will have a value of 0 and we don't want that.
     @api.depends('previous_statement_id', 'previous_statement_id.balance_end_real')
     def _compute_starting_balance(self):
-        for statement in self:
+        # When a bank statement is inserted out-of-order several fields needs to be recomputed.
+        # As the records to recompute are ordered by id, it may occur that the first record
+        # to recompute start a recursive recomputation of field balance_end_real
+        # To avoid this we sort the records by date
+        for statement in self.sorted(key=lambda s: s.date):
```

Comment:
Hi! The statements are sorted by `date desc, name desc, id desc` by default, so why not keep the same order here?
```python
        for statement in self.sorted(reverse=True):
```